pool/session.go

```diff
@@ -72,10 +72,13 @@ func (p *Pool) withinContainerSession(
 		return nil
 	}
 
-	cacheKey := formCacheKey(fmt.Sprintf("%p", c), signer)
+	cacheKey := cacheKeyForSession(fmt.Sprintf("%p", c), signer, verb, containerID)
 
 	tok, ok := p.cache.Get(cacheKey)
 	if !ok {
+		tok.ForVerb(verb)
+		tok.BindContainer(containerID)
+
 		// init new session
 		err = initSession(ctx, &tok, c, p.stokenDuration, signer)
```

Comment:
Maybe this suggests we need _two_ caches though. One for complete tokens and one for remote session keys (to avoid network requests when they're not needed).